src/sdp/SDP.ts

```diff
@@ -658,4 +659,169 @@ export default class SDP {
+  private sectionWithCodecPreferences(
+    section: string,
+    preferences: VideoCodecCapability[]
+  ): string {
+    const codecNamesToPayloadTypes: Map<string, string> = new Map();
+    const lines = SDP.splitLines(section);
+
+    // First we get the payload types and their respective `a=rtpmap` lines for our provided preferences
+    lines.forEach(line => {
+      if (!/^a=rtpmap:/.test(line)) {
+        return;
+      }
+      for (const preference of preferences) {
+        // Check if theres a match for the encoding name and clock rate as defined in 'RFC 4566 Section 6':
+        // a=rtpmap:<payload type> <encoding name>/<clock rate> [/<encoding parameters>]
+        // E.g. 'a=rtpmap:125 H264/90000'
+        if (!line.includes(`${preference.codecName}/${preference.codecCapability.clockRate}`)) {
+          continue;
+        }
+        const payloadMatch = /^a=rtpmap:([0-9]+)\s/.exec(line); // Get the payload type
+
+        // We may need to check other parameters (e.g. fmtp line) in addition to the codec name
+        let codecMatches = false;
+        if (preference.codecCapability.sdpFmtpLine !== undefined) {
+          // Check the fmtp line
+          for (const prospectiveFmtpLine of lines) {
+            if (
+              prospectiveFmtpLine.startsWith(
+                `a=fmtp:${payloadMatch[1]} ${preference.codecCapability.sdpFmtpLine}`
+              )
+            ) {
+              codecMatches = true;
+              break;
+            }
+          }
+        } else {
+          // No 'fmtp' line, nothing else to check
+          codecMatches = true;
+        }
+
+        if (codecMatches) {
+          codecNamesToPayloadTypes.set(preference.codecName, payloadMatch[1]);
+          break;
+        }
+      }
+    });
+
+    // RFC 4566 5.14
+    // When a list of payload type numbers is given, this implies that all of these
+    // payload formats MAY be used in the session, but the first of these
+    // formats SHOULD be used as the default format for the session.
+
+    const payloadTypesToRemove: Set<string> = new Set(codecNamesToPayloadTypes.values());
+    // Remove payloads from the media line. m=video 9 UDP/+++ <payload> <payload> <payload> ...
+    const mline = lines[0].split(' ').filter((text: string) => !payloadTypesToRemove.has(text));
+    // Then splice them back in, in preferred order at the start of the list
+    const orderedPreferedPayloadTypes = Array.from(codecNamesToPayloadTypes.values()).sort(
+      (name1: string, name2: string) => {
+        const priority1 = preferences.findIndex(capability => {
+          return codecNamesToPayloadTypes.get(capability.codecName) === name1;
+        });
+        const priority2 = preferences.findIndex(capability => {
+          return codecNamesToPayloadTypes.get(capability.codecName) === name2;
+        });
+        return priority1 - priority2;
+      }
+    );
+    mline.splice(3, 0, ...orderedPreferedPayloadTypes.values());
```

Comment:
Could we put a comment on why it starts from 3 :D You have something similar in the function below.